### DISCUSSION TITLE
Will HomeSpan devices integrate with Matter?

### DISCUSSION THREAD

**Discussion by tysonmatanich** [HomeSpan/HomeSpan#247] [General]
With Matter (formerly Project CHIP) likely coming out sometime this year, do we know how or if HomeSpan will work with it?

https://buildwithmatter.com/
https://github.com/project-chip/connectedhomeip

**Comment by HomeSpan** [comment]
According to Apple, the HomeKit protocol will continue to be used - Apple will adopt Matter in parallel.  As long as HomeKit is supported, HomeSpan will continue to work.  However, the reverse is not true - HomeSpan cannot be used with a system that expects the Matter protocol.  Once Matter gets underway my guess is that there will be many Matter libraries we will be able to use on the ESP32.  I'm looking forward to seeing how this all works in practice.

From my perspective, the biggest challenge with HomeKit (and possibly Matter) is the lack of integration between the interface and the device.  I'm just finishing up the next release of HomeSpan (1.4.3) with a focus on addressable LEDs (NeoPixels from Adafruit, and equivalent).  One of my sketches implements a number of special effects (blinking lights, the Knight-Rider effect, etc).  There is unfortunately no easy way to "select" these from the Home App.  I've been using the Eve for HomeKit app on the iPhone as a supplement since it will create a generic interface for any Characteristic it does not recognize.  Creating a custom Characteristic is easy in HomeSpan - I created one with 5 valid values, and now I can use the Eve App to select which effect to turn on. 

But this is very hacky.  What I would rather do is embed the interface logic into my custom Characteristic so the the Home App would render a "custom control."  I'm toying with the idea of creating such an app, but with Matter coming out at some point, all of this may be superseded in the future.
